src/Compilers/Core/CodeAnalysisTest/ShadowCopyAnalyzerAssemblyLoaderTests.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit;
+
+// The ShadowCopyAnalyzerAssemblyLoader type is only defined for the below platforms
+#if NET472 || NETCOREAPP1_1 || NETCOREAPP2_1 
+
+namespace Microsoft.CodeAnalysis.UnitTests
+{
+    public sealed class ShadowCopyAnalyzerAssemblyLoaderTests : TestBase
+    {
+        [Fact, WorkItem(32226, "https://github.com/dotnet/roslyn/issues/32226")]
+        public void LoadWithDependency()
+        {
+            var directory = Temp.CreateDirectory();
+            var immutable = directory.CopyFile(typeof(ImmutableArray).Assembly.Location);
+            var microsoftCodeAnalysis = directory.CopyFile(typeof(DiagnosticAnalyzer).Assembly.Location);
+
+            var analyzerDependencyFile = CreateAnalyzerDependency();
+            var analyzerMainFile = CreateMainAnalyzerWithDependency(analyzerDependencyFile);
+            var loader = new ShadowCopyAnalyzerAssemblyLoader(Path.Combine(directory.Path, "AnalyzerAssemblyLoader"));
+
+            var analyzerMainReference = new AnalyzerFileReference(analyzerMainFile.Path, loader);
+            analyzerMainReference.AnalyzerLoadFailed += (_, e) => AssertEx.Fail(e.Exception.Message);
+            var analyzerDependencyReference = new AnalyzerFileReference(analyzerDependencyFile.Path, loader);
+            analyzerDependencyReference.AnalyzerLoadFailed += (_, e) => AssertEx.Fail(e.Exception.Message);
+
+            var analyzers = analyzerMainReference.GetAnalyzersForAllLanguages();
```

Comment:
Verified that the test fails prior to the product code change with `Message: Could not load file or assembly 'AnalyzerDependency, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. The system cannot find the file specified`